lib/TWCManager/Policy/Policy.py

```diff
@@ -128,6 +128,10 @@ def __init__(self, master):
                 if policy_engine.get("policyCheckInterval"):
                     self.policyCheckInterval = policy_engine.get("policyCheckInterval")
 
+    def applyPolicyImmediately(self):
+        self.lastPolicyCheck = 0
+        self.setChargingPerPolicy()
```

Comment:
This might or might not be necessary, since the next slave heartbeat would cause the evaluation to happen anyway, but there's no extra work since it will immediately update the timestamp and future runs will skip on that basis.  Is there a threading issue with doing this on the thread handling the HTTP/MQTT requests?

Reply:
That's true, there's no thread safety for the ```lastPolicyCheck``` variable, meaning the ```lastPolicyCheck``` timestamp could be updated before we call ```setChargingPerPolicy``` if there was already a ```setChargingPerPolicy``` call in progress when we call it again. However if that did happen it would be imperceptible to the user because we'd already be re-evaluating policy (and that variable is only updated by the ```setChargingPerPolicy``` function itself), so while it is a potential race condition I think it's probably one that would give us the intended outcome - ie not running the ```setChargingPerPolicy()``` function twice even if called while a previous function were in progress.